### PR TITLE
Problem: support SSE generic message

### PR DESCRIPTION
**Pull request by jeffdegott**
Solution: Do it.
Handle SSE generic message on STREAM-DELIVER stream, in addition of the fty-proto ALERTS and ASSETS messages. A generic message is composed of three frames (in order): topic, jsonPayload and assetID (3rd is optional). See sse.cc:changeSseMessage2Json().

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,6 +32,7 @@ src/fty_rest_selftest
 *.out
 *.app
 core
+go.sh
```
Comment (jimklimov) [inline .gitignore]: This is ignored twice :)

File: ci_build.sh [hunk 2]
```diff
@@ -4,17 +4,9 @@
 #  THIS FILE IS 100% GENERATED BY ZPROJECT; DO NOT EDIT EXCEPT EXPERIMENTALLY  #
 #  READ THE ZPROJECT/README.MD FOR INFORMATION ABOUT MAKING PERMANENT CHANGES. #
 ################################################################################
-# NOTE: This script was customized (PATH, added commented-away quicktest lines,#
```
Comment (jimklimov) [inline ci_build.sh]: Project regeneration did not retain customizations (see our doc on proper procedure to mix the two worlds), and seems to use an older version of zproject templates. In short, using `git difftool` is essential to review codebase changes after regenerating.

File: src/libfty_rest.pc.in [hunk 3]
```diff
@@ -12,7 +12,7 @@ Name: libfty_rest
 Description: Common core REST API for 42ity project
 Version: @VERSION@
 
-Requires:@pkgconfig_name_libsodium@ @pkgconfig_name_libzmq@ @pkgconfig_name_libczmq@ @pkgconfig_name_libcidr@ @pkgconfig_name_cxxtools@ @pkgconfig_name_tntnet@ @pkgconfig_name_tntdb@ @pkgconfig_name_log4cplus@ @pkgconfig_name_libfty_common_logging@ @pkgconfig_name_libfty_common@ @pkgconfig_name_libfty_common_db@ @pkgconfig_name_libfty_common_rest@ @pkgconfig_name_libmlm@ @pkgconfig_name_openssl@ @pkgconfig_name_libfty_common_mlm@ @pkgconfig_name_libsasl2@ @pkgconfig_name_libfty_proto@
+Requires:@pkgconfig_name_libsodium@ @pkgconfig_name_libzmq@ @pkgconfig_name_libczmq@ >= 3.0.2 @pkgconfig_name_libcidr@ @pkgconfig_name_cxxtools@ @pkgconfig_name_tntnet@ @pkgconfig_name_tntdb@ @pkgconfig_name_log4cplus@ @pkgconfig_name_libfty_common_logging@ @pkgconfig_name_libfty_common@ @pkgconfig_name_libfty_common_db@ @pkgconfig_name_libfty_common_rest@ @pkgconfig_name_libmlm@ >= 1.0.0 @pkgconfig_name_openssl@ @pkgconfig_name_libfty_common_mlm@ @pkgconfig_name_libsasl2@ @pkgconfig_name_libfty_proto@ >= 1.0.0
```
Comment (jimklimov) [inline src/libfty_rest.pc.in]: Older zproject template in play.

File: packaging/redhat/fty-rest.spec [hunk 4]
```diff
@@ -1,6 +1,5 @@
 #
 #    fty-rest - Common core REST API for 42ity project
-#    NOTE: This file was customized after generation, be sure to keep it
```
Comment (jimklimov) [inline packaging/redhat/fty-rest.spec]: Customizations required and not retained.

File: packaging/debian/libfty-rest1.install [hunk 5]
```diff
@@ -1,6 +1 @@
-# Note: this file is customized after zproject generation, be sure to keep it
```
Comment (jimklimov) [inline packaging/debian/libfty-rest1.install]: Customizations required and not retained.

File: packaging/debian/libfty-rest-dev.install [hunk 6]
```diff
@@ -1,9 +1,3 @@
-# Note: this file is customized after zproject generation, be sure to keep it
```
Comment (jimklimov) [inline packaging/debian/libfty-rest-dev.install]: Customizations required and not retained.

File: packaging/debian/control [hunk 7]
```diff
@@ -1,6 +1,5 @@
 #
 #    fty-rest - Common core REST API for 42ity project
-# Note: this file is customized after zproject generation, be sure to keep it
```
Comment (jimklimov) [inline packaging/debian/control]: Customizations required and not retained.

File: configure.ac [hunk 8]
```diff
@@ -177,6 +177,7 @@ AS_IF([test x"${search_libsodium}" = xyes], [
     PRE_SEARCH_LIBS="${LIBS}"
 
     found_pkgconfig=""
+    found_linkname=""
```
Comment (jimklimov) [inline configure.ac]: Older zproject template in play.

File: src/web/src/sse.cc [hunk 9]
```diff
@@ -366,6 +382,41 @@ std::string Sse::changeFtyProtoAsset2Json(fty_proto_t *asset)
...
+  if (!assetID.empty()
+    && (_assetsOfDatacenter.find(assetID) == _assetsOfDatacenter.end())
+  )
+  {
+    //NOTE: here, message is (partially) consumed and can't be handled elsewhere on return
+    log_debug("skipping due to element_src '%s' not being in the list", assetID.c_str());
+    return "";
+  }
+
+  return "data:{\"topic\":\"" + topic + "\",\"payload\":" + jsonPayload + "}\n\n";
```
Comment (jimklimov) [inline src/web/src/sse.cc]: Shouldn't `"data"` be quoted?

File: src/web/src/admin_sse.ecpp [hunk 10]
```diff
@@ -163,91 +167,117 @@ bool database_ready;
...
+        if (which == sseManager.getPipe())
             recv_msg = sseManager.getMessageFromMlm();
-        }
-        if (!recv_msg ){
+
+        if (!recv_msg)
+        {
             if (zpoller_expired (sseManager.getPoller())) {
                 //Send heartbeat message
                 json = "data:{\"topic\":\"heartbeat\",\"payload\":{}}\n\n";
```
Comment (jimklimov) [inline src/web/src/admin_sse.ecpp]: Shouldn't `"data"` be quoted?

File: src/web/src/admin_sse.ecpp [hunk 11]
```diff
@@ -163,91 +167,117 @@ bool database_ready;
...
             log_info ("sse : Token revoked or expired");
             break;
         }
 
         // Each minute, send the time before expiration
-        if (now - sendNextExpTime > 60000) {
+        if ((now - sendNextExpTime) > 60000)
+        {
             // send the time before the token is expired
             json = "data:{\"topic\":\"session\",\"payload\":{\"exptime\":" ;
```
Comment (jimklimov) [inline src/web/src/admin_sse.ecpp]: Shouldn't `"data"` be quoted?

File: ci_build.sh (hunk 2 above)
Comment (jeffdegott) [reply ci_build.sh]: Seems to be fixed w/ zproject upgrade...

File: src/web/src/sse.cc (hunk 9 above)
Comment (jeffdegott) [reply src/web/src/sse.cc]: Checked. Data isn't quoted.

**Pull request closed without merging**